src/net_processing.cpp

```diff
@@ -324,6 +324,18 @@ struct Peer {
     /** Work queue of items requested by this peer **/
     std::deque<CInv> m_getdata_requests GUARDED_BY(m_getdata_requests_mutex);
 
+    struct CompareCBlockIndexRef {
+        bool operator()(const CBlockIndex& a, const CBlockIndex& b) const { return &a < &b; }
+    };
+    /** Protects m_known_chain_tips */
+    Mutex m_known_chain_tips_mutex;
+    /**
+     * Chain tips that this peer knows we have because they send them to us.
+     * Limited to the number of chain tips in our global block index.
+     */
+    std::set<std::reference_wrapper<const CBlockIndex>, CompareCBlockIndexRef>
```

Comment:
Would it be better to track chain tips per-network rather than per-peer? Is it a privacy leak if one peer on a network can determine that the node received a stale block header from a different peer on the same network?

Reply:
Does it mean that the attack will be still possible for a tor-only node with one bind=onion?

Reply:
Could you elaborate how you think the attack would be possible in that case?